### PR TITLE
Issue 151 - Image vector search works !!

### PR DESCRIPTION
**Pull request by atarora**
1. Fixed the data encoding process to use the image encoding capability of the clip model
2. Improved the exception handling in the encoding process
3. Updated dependencies and requirements
4. Changes to the encoding service
5. Clip model is not saved locally as there is no provision

### CODE REVIEW

File: embeddings/app/clipL14/loadModel.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import torch
+import torchvision.transforms as transforms
+import clip
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+model, preprocess = clip.load('ViT-L/14', device)
+
+#model.save('/code/app/clip-ViT-L-14.model')
+# Running into AttributeError: 'CLIP' object has no attribute 'save'
+# No save method at https://github.com/openai/CLIP
```
Comment (renekrie) [inline embeddings/app/clipL14/loadModel.py:12]: Is this file being called? - Can't see it in the Dockerfile

File: data-encoder/ecommerce/vectors/query_vector.py
Comment (renekrie) [inline data-encoder/ecommerce/vectors/query_vector.py:1]: Do we need the commented out code? - otherwise we could just revert all changed in the file

**Review by renekrie — CHANGES_REQUESTED** [review]
Please see comments

File: data-encoder/ecommerce/vectors/query_vector.py
Comment (atarora) [reply data-encoder/ecommerce/vectors/query_vector.py:1]: I kept the file with the reason why this wasn't possible and to maintain the existing structure.
This is intended and not a mistake.

File: embeddings/app/clipL14/loadModel.py (hunk 1 above)
Comment (atarora) [reply embeddings/app/clipL14/loadModel.py:12]: No !

File: data-encoder/ecommerce/vectors/query_vector.py
Comment (renekrie) [reply data-encoder/ecommerce/vectors/query_vector.py:1]: Could you please leave a note there à la "This doesn't work" and indicate where it fails - as it is it just looks like code that could be removed (which someone might just do in the future).

File: embeddings/app/clipL14/loadModel.py (hunk 1 above)
Comment (renekrie) [reply embeddings/app/clipL14/loadModel.py:12]: Could you please remove the file then?

File: data-encoder/ecommerce/vectors/query_vector.py
Comment (atarora) [reply data-encoder/ecommerce/vectors/query_vector.py:1]: I think I was mistaken , I sort of clubbed the 2 questions together.
The query.py is for testing purposes to generate encoding using different transformer models.
The commented out code is an example of using other transformer model.

All of the code works :) although this file has more of the checkpoint feature.

File: embeddings/app/clipL14/loadModel.py (hunk 1 above)
Comment (atarora) [reply embeddings/app/clipL14/loadModel.py:12]: The answer added below was actually intended for this comment : 
"I kept the file with the reason why this wasn't possible and to maintain the existing structure."

Comment (atarora) [reply embeddings/app/clipL14/loadModel.py]: done

File: data-encoder/ecommerce/vectors/query_vector.py
Comment (atarora) [reply data-encoder/ecommerce/vectors/query_vector.py:1]: Reworked and added comments for the designated usage.

**Review by renekrie — APPROVED** [review]
Looks good to me! Thank you @atarora !

**Review by epugh — APPROVED** [review]

**Pull request merged**